rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java

```diff
@@ -96,20 +97,24 @@ public void execute(ApplicationContext currentContext) throws CLIException {
 
     }
 
-    private void validateFilePath(ApplicationContext currentContext) {
+    private void validateFilePath(ApplicationContext currentContext) throws JobCreationRestException {
+
         if (!isFilePathValid(pathname)) {
             throw new CLIException(REASON_INVALID_ARGUMENTS, String.format("'%s' is not a valid file.", pathname));
         }
 
-        if (!isFileExisting(pathname)) {
-            throw new CLIException(REASON_INVALID_ARGUMENTS, String.format("'%s' does not exist.", pathname));
+        if (!isValidFileMimeType(pathname)) {
+            throw new JobCreationRestException("Unknown job descriptor type: " + pathname);
         }
 
-        if (isFileEmpty(pathname)) {
-            writeLine(currentContext, "File " + pathname + " is empty ");
+        if (isFileExisting(pathname) && isFileEmpty(pathname)) {
             throw new CLIException(REASON_FILE_EMPTY, String.format("'%s' is empty.", pathname));
         }
 
+        if (!isFileExisting(pathname)) {
```

Comment:
you can move this check before isFileEmpty(pathname). and then only check if the file is empty. It will simplify the code